Nightly build errors intel/17 and intel/18 with Pthread,Serial: Kokkos_ThreadsExec.cpp(553): error: const variable "sentinel" requires an initializer

Changes possibly from #2715 are breaking intel/17.1.132 and 18.0.0.20170811  builds in the nightlies on Bowman (KNL arch) and intel/18.1.163 on Blake (Skylake)

**Error message**
```
[  6%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/Threads/Kokkos_ThreadsExec_base.cpp.o
cd /home/jenkins/bowman/workspace/Kokkos_SLURM_inner_test@2/20200219025302/core/src && /home/projects/x86-64-knl/intel/compilers/2018/compilers_and_libraries_2018.0.128/linux/bin/intel64/icpc   -I/home/jenkins/bowman/workspace/Kokkos_SLURM_inner_test@2/20200219025302/core/src -I/home/jenkins/bowman/workspace/Kokkos_SLURM_inner_test@2/core/src -I/home/jenkins/bowman/workspace/Kokkos_SLURM_inner_test@2/20200219025302  -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized -O3 -DNDEBUG   -xMIC-AVX512 -pthread -std=c++11 -o CMakeFiles/kokkoscore.dir/Threads/Kokkos_ThreadsExec_base.cpp.o -c /home/jenkins/bowman/workspace/Kokkos_SLURM_inner_test@2/core/src/Threads/Kokkos_ThreadsExec_base.cpp
/home/jenkins/bowman/workspace/Kokkos_SLURM_inner_test@2/core/src/Threads/Kokkos_ThreadsExec.cpp(553): error: const variable "sentinel" requires an initializer -- class "Kokkos::Impl::<unnamed>::Sentinel" has no user-provided default constructor
    static const Sentinel sentinel;
                                  ^

compilation aborted for /home/jenkins/bowman/workspace/Kokkos_SLURM_inner_test@2/core/src/Threads/Kokkos_ThreadsExec.cpp (code 2)
make[2]: *** [core/src/CMakeFiles/kokkoscore.dir/Threads/Kokkos_ThreadsExec.cpp.o] Error 2
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory `/home/jenkins/bowman/workspace/Kokkos_SLURM_inner_test@2/20200219025302'
make[1]: *** [core/src/CMakeFiles/kokkoscore.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

--- a/core/src/Threads/Kokkos_ThreadsExec.cpp
+++ b/core/src/Threads/Kokkos_ThreadsExec.cpp
@@ -550,7 +550,8 @@ int ThreadsExec::is_initialized() { return nullptr != s_threads_exec[0]; }
 void ThreadsExec::initialize(unsigned thread_count, unsigned use_numa_count,
                              unsigned use_cores_per_numa,
                              bool allow_asynchronous_threadpool) {
-  static const Sentinel sentinel;
+  // need to provide an initializer for Intel compilers
+  static const Sentinel sentinel = {};
 
   const bool is_initialized = 0 != s_thread_pool_size[0];
 

